discovery/gossiper.go

```diff
@@ -3489,6 +3489,30 @@ func (d *AuthenticatedGossiper) handleAnnSig(ctx context.Context,
 			peerID := nMsg.source.SerializeCompressed()
 			log.Debugf("Got AnnounceSignatures for channel with " +
 				"full proof.")
+			syncer, ok := d.syncMgr.GossipSyncer(
+				route.Vertex(peerID),
+			)
+			if !ok {
+				log.Errorf("Failed to get gossip syncer for "+
+					"peer=%x", peerID)
+				return nil, false
+			}
+
+			// If we already sent our proof to this peer once since
+			// (re)connecting, then we can skip sending it again.
+			syncer.Lock()
+			alreadySent := syncer.proofSentToChan.Contains(
+				ann.ChannelID,
+			)
+			syncer.Unlock()
+			if alreadySent {
+				log.Debugf("Skipping sending announcement " +
+					"signatures since we already did " +
```

Comment:
i think it makes more sense to ask the `peer` that we are handed here the question of "have we sent this during this connection lifetime yet?"